tests/test_utility.py

```diff
@@ -17,20 +17,26 @@
 
 
 def test_show_version():
-    with pytest_warns_bounded(
-        UserWarning,
-        match="Setuptools is replacing distutils",
-        upper="3.11.99",
-        version_str=platform.python_version(),
-    ):
-        context: AbstractContextManager
-        if PD_LTE_22 and NUMPY20:  # https://github.com/PyTables/PyTables/issues/1172
-            context = pytest.raises(ValueError)
-        else:
-            context = nullcontext()
-        with context:
-            check(assert_type(pd.show_versions(True), None), type(None))
-            check(assert_type(pd.show_versions(False), None), type(None))
+    """Test show_versions method types with split case for pandas and python versions."""
```

Comment:
If you set `context = nullcontext()` here, then you don't need the `else`, and  you can have the 2 lines for the test as `with context` where `with context` is at the level of `if` and you avoid the repeated test.